tests/path/win32.c

```diff
@@ -186,6 +202,8 @@ void test_path_win32__canonicalize(void)
 	test_canonicalize(L"\\\\server\\\\share\\\\foo\\\\bar", L"\\\\server\\share\\foo\\bar");
 	test_canonicalize(L"\\\\server\\share\\..\\foo", L"\\\\server\\foo");
 	test_canonicalize(L"\\\\server\\..\\..\\share\\.\\foo", L"\\\\server\\share\\foo");
+
+	test_path_git_win32__canonicalize_path(L"\\\\?\\UNC\\server\\C$\\folder", L"\\\\server\\C$\\folder");
```

Comment:
All the above tests check whether the function `git_win32_path_canonicalize` works, whereas the new call checks whether `git_win32__canonicalize_path` functions as expected. I'd thus propose to move it into a new unit test (e.g. a function `void test_path_win32__canonicalize_path(void)`).

Our existing API really is a bit confusing. `git_win32_path_canonicalize` and `git_win32__canonicalize_path` sound like they should do the same thing. But the former mostly sanitizes slashes, while the latter will strip leading namespace prefixes. We should probably rename the second one, but that needn't be part of this pull request.

Reply:
Yep. Makes sense. Done.